hawkular-command-gateway/hawkular-command-gateway-war/src/main/java/org/hawkular/cmdgw/command/bus/BusEndpointProcessors.java

```diff
@@ -222,8 +221,8 @@ protected void onBasicMessage(BasicMessageWithExtraData<BasicMessage> messageWit
     @Inject
     private BusCommandContextFactory commandContextFactory;
 
-    @Resource(name = Constants.CONNECTION_FACTORY_JNDI)
-    private ConnectionFactory connectionFactory;
+    @Inject
+    private BusConnectionFactoryProvider connectionFactoryProvider;
```

Comment:
You could actually inject the ConnectionFactory directly, using a CDI qualifier, like, `@Inject @Ready ConnectionFactory connectionFactory` and have your `BusConnectionFactoryProvider#getConnectionFactory` to be a `@Produces @Ready`. 

But please, use a better name than `@Ready` :)

The qualifier can be something similar to this: https://github.com/hawkular/hawkular-accounts/blob/master/api/src/main/java/org/hawkular/accounts/api/NamedSetting.java